pkg/datasources/users.go

```diff
@@ -107,49 +106,47 @@ func Users() *schema.Resource {
 
 func ReadUsers(d *schema.ResourceData, meta interface{}) error {
 	db := meta.(*sql.DB)
+	ctx := context.Background()
+	client := sdk.NewClientFromDB(db)
+
 	userPattern := d.Get("pattern").(string)
 
-	account, err := snowflake.ReadCurrentAccount(db)
-	if err != nil {
+	account, err1 := client.ContextFunctions.CurrentAccount(ctx)
```

Comment:
wait we already have CurrentAccount? Then what was the previous Current()? I don't see that anywhere in here: https://docs.snowflake.com/en/sql-reference/functions-context

Reply:
The problem is, we've already had a family of `CurrentX()` functions. In the old snowflake package there was one method, fetching both account and region. To spare two invocation, I created a new one in the SDK, which fetches all these CurrentX params. We can name it differently if you have an idea how. CurrentAll? CurrentAccountAll? any ideas? cc: @sfc-gh-jcieslak

Reply:
if your concern is with multiple invocations, there is a way we can fix that by batching requests together. I don't know exactly how we would accomplish this, but it is worth thinking about.

Reply:
It was not a concern about the multiple invocations performance but the need to check the errors for each. Now, we fetch multiple values together, and I think it's easier to use it.